superset/views/core.py

```diff
@@ -1237,9 +1238,12 @@ def explore_json(self, datasource_type=None, datasource_id=None):
         requests that GETs or POSTs a form_data.
 
         `self.generate_json` receives this input and returns different
-        payloads based on the request args in the first block
+        payloads based on the request args in the first block"""
 
-        TODO: break into one endpoint for each return shape"""
+        if request.method == 'GET' and not is_feature_enabled('CLIENT_CACHE'):
```

Comment:
```suggestion
        if request.method != 'POST' and not is_feature_enabled('CLIENT_CACHE'):
```
^ this seems slightly better since we really only want to allow `POST`s (and requiring more explicit-ness seems like a good thing if that were to change in the future)

Reply:
fixed.